src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,200 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public T unlink(Node<T> x) {
+        T element = x.element;
+        Node<T> next = x.next;
+        Node<T> prev = x.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            x.prev = null;
+        }
+
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            x.next = null;
+        }
+
+        x.element = null;
+        size--;
+        return element;
+    }
+
+    public void linkBefore(T e, Node<T> node) {
+        final Node<T> pred = node.prev;
+        final Node<T> newNode = new Node<>(pred, e, node);
+        node.prev = newNode;
+        if (pred == null) {
+            first = newNode;
+        } else {
+            pred.next = newNode;
+        }
+        size++;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> tmp = last;
+        Node<T> newNode = new Node<>(last, value, null);
+        last = newNode;
+        if (tmp == null) {
+            first = newNode;
+        } else {
+            tmp.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            linkBefore(value, getNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
```

Comment:
Use a simple loop and for every single element call method `add(T value)`. You've already had a logic for adding nodes there